src/calliope/config/model_def_schema.yaml

```diff
@@ -254,6 +264,63 @@ properties:
           - $ref: "#/$defs/unIndexedParamVal"
           - $ref: "#/$defs/IndexedParam"
 
+  data_sources:
+    type: array
+    description: >-
+      Reference to files from which to load parts (or all) of the model definition.
+      Loaded data will be _overridden_ by any data provided in the YAML definitions.
+      Conflicting data between loaded files will be resolved by overriding in the order the data is provided.
+      That is, data in list item `N` takes precedence over data in list item `N-1`.
+    items:
```

Comment:
I'm slightly hesitant since we do so much internal processing of the loaded data that anything passed to pandas kwargs might break things easily. But it would be easy enough to include...